api/admin/controller/__init__.py

```diff
@@ -1561,7 +1563,10 @@ def _get_prior_test_results(self, item, protocol_class=None):
                 self_test_results = ExternalSearchIndex.prior_test_results(
                     self._db, None, self._db, item
                 )
-
+            elif self.type == "metadata service":
+                self_test_results = protocol_class.prior_test_results(
+                    self._db, *extra_args
+                )
```

Comment:
I'm envisioning a future where `_get_prior_test_results` can just call `find_protocol_class` to get the arguments, and pass those arguments into `prior_test_results`. This little bit would become basically the whole method.